activerecord/lib/active_record/persistence.rb

```diff
@@ -563,7 +563,19 @@ def destroy(id)
       #   # Delete multiple rows
       #   Todo.delete([2,3,4])
       def delete(id_or_array)
-        delete_by(primary_key => id_or_array)
+        case id_or_array
+        when nil
+          0
+        when Array
+          ids = id_or_array.compact
```

Comment:
I think the compact is a bit much. I'm fine with just checking `nil` and empty arrays though, as I see how we can end up here on the happy path.

Reply:
Yeah, makes sense! Changed that.